library/ssl_srv.c

```diff
@@ -3585,15 +3585,19 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl )
     /* Read the message without adding it to the checksum */
     do {
 
-        if( ( ret = mbedtls_ssl_read_record_layer( ssl ) ) != 0 )
+        do ret = mbedtls_ssl_read_record_layer( ssl );
+        while( ret == MBEDTLS_ERR_SSL_CONTINUE_PROCESSING );
```

Comment:
@hanno-arm you are correct, my mistake regarding the networking part. However, is it possible we will endless loop on invalid record? I mean, `mbedtls_ssl_read_record_layer` will call `mbedtls_ssl_parse_record` which will fail with `MBEDTLS_ERR_SSL_INVALID_RECORD` error to its caller (`mbedtls_ssl_read_record_layer`), which  will return  `MBEDTLS_ERR_SSL_CONTINUE_PROCESSING `.